### PR TITLE
Remove Python 3.3 check in mock.pyi

### PR DESCRIPTION
**Pull request by mxr**
See context at https://github.com/python/typeshed/pull/2173#issuecomment-394819364

I fixed the comment copied from `unittest/mock.pyi` and kept the `3.7` check - let me know if that should go

cc @JelleZijlstra @rowillia

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
The 3.7 check is fine. You should apply the same change to unittest/mock.pyi; a test will fail otherwise.

**Comment by mxr** [comment]
@JelleZijlstra done

The files are pretty much identical though

```diff
$ diff -u third_party/2and3/mock.pyi stdlib/3/unittest/mock.pyi
--- third_party/2and3/mock.pyi	2018-06-05 15:24:33.000000000 -0400
+++ stdlib/3/unittest/mock.pyi	2018-06-05 17:16:35.000000000 -0400
@@ -1,4 +1,4 @@
-# Stubs for mock
+# Stubs for unittest.mock

 import sys
 from typing import Any, Optional, Type
```

In the future would it be possible to use a symlink or some automation to help keep them in sync?

File: third_party/2and3/mock.pyi [hunk 1]
```diff
@@ -1,181 +1,180 @@
-# Stubs for unittest.mock
+# Stubs for mock
```
[the remark this answers is not in the document]
Comment (mxr) [reply third_party/2and3/mock.pyi]: Sure

**Comment by mxr** [comment]
@JelleZijlstra gotcha, by automation I meant a `pre-commit` hook which would `cp third_party/2and3/mock.pyi stdlib/3/unittest/mock.pyi` (or something)

**Comment by JelleZijlstra** [comment]
That seems risky because we don't know in what direction we should cp.

**Comment by mxr** [comment]
Perhaps you could use `mtime` to decide which file to be the source of truth? Anyways, not trying to derail this PR, was just a thought about simplifying this edge case. Thanks for looking at the PR!

**Comment by JelleZijlstra** [comment]
I guess that could work, but I'm not sure the additional complexity is worth it. We might accept it if somebody implements it though. I'd be worried about a case where somebody modifies both files and we overwrite their work in one, but I suppose we could check which file(s) are dirty in git.

Anyway, we can just have this conversation while we wait for Travis to finish. :) Once it's done I'll merge this.

**Comment by mxr** [comment]
Gotcha. Well for 17 files it's probably not a big deal. Thanks again

**Pull request merged**